Change GPG home directory configuration

jrwdunham opened the pull request:
- If set, the environment variable GNUPG_HOME_PATH will determine where GPG
  writes to disk.
- If the GNUPG_HOME_PATH env var is undefined, GPG will use the absolute path
  of the SS's internal location.
- Fixes #271

sevein commented on storage_service/storage_service/settings/base.py:
```diff
@@ -405,3 +405,5 @@ def is_true(env_str):
 except ValueError:
     BAG_VALIDATION_NO_PROCESSES = 1
+
+GNUPG_HOME_PATH = environ.get('GNUPG_HOME_PATH', None)
```
Shouldn't be prefixed with `SS_` like the others?

sevein commented on storage_service/common/gpgutils.py:
```diff
@@ -40,23 +35,42 @@
 ENCR_FAILS = 'no'
 
 
+class GPG(object):
```
The linter didn't like your style.

sevein commented on storage_service/common/gpgutils.py:
```diff
@@ -40,23 +35,42 @@
+        return self._gpg
+
+
+gpg = GPG()
```
Singletons are good for you.

sevein reviewed (changes requested):
I've added a couple of observations.

jrwdunham committed:
Change GPG home directory configuration

- If set, the environment variable GNUPG_HOME_PATH will determine where GPG
  writes to disk.
- If the GNUPG_HOME_PATH env var is undefined, GPG will use the absolute path
  of the SS's internal location.
- Fixes #271
